src/main/webapp/edit-recipe.html

```diff
@@ -0,0 +1,59 @@
+<!DOCTYPE html>
+<html>
+
+<head>
+  <meta charset="UTF-8">
+  <title>Edit Recipe</title>
+  <script type="text/javascript" src="script.js"></script>
+</head>
+
+<body>
+  <h1>Edit Recipe!</h1>
+
+  <form action="/new-recipe" method="POST">
+    <h2>Name</h2>
+    <label for="name">What's your recipe called?</label><br/>
+    <textarea id="name" name="name" rows="1"></textarea><br/><br/>
+
+    <h2>Description</h2>
+    <label for="description">Give your recipe some more context.</label><br/>
+    <textarea id="description" name="description" rows="3"></textarea><br/><br/>
+
+    <h2>Tags</h2>
+    <div id="tags">
+      <label for="tag1">Tag 1</label></br>
+      <textarea id="tag1" name="tag1" rows="1"></textarea></br></br>
+      <label for="tag2">Tag 2</label></br>
+      <textarea id="tag2" name="tag2" rows="1"></textarea></br></br>
+    </div>
+
+    <h2>Ingredients</h2>
+    <div id="ingredients">
+      <label for="ingredient1">Ingredient 1</label></br>
+      <textarea id="ingredient1" name="ingredient1" rows="1"></textarea></br></br>
```

Comment:
Can't you instead have an ordered list here? It might be much more convenient, in terms of displaying any number of ingredients/steps and also accessing them here and in the servlet.

Reply:
In the branches I mention in the above comment, I use custom HTML elements to easily allow any number of parameters. I prefer this over an ordered list because eventually, ingredients and steps will have multiple different properties in addition to string descriptions, so a custom element will allow me to cleanly organize their properties into a reusable template.